### ISSUE TITLE
remove sensitive data from oso-proxy logs

### ISSUE THREAD

**Issue by nurali-techie** [openshiftio/openshift.io#3882]
Currently, oso-proxy (i.e. traefik) logs `"Authorization"` header and `"access_token"` parameter into logs.  Till traefik 1.5.x it was not possible to remove this fields from logs.  With Traefik 1.6.x they have provided access_log_filtering feature [link](https://docs.traefik.io/v1.6/configuration/logs/#access-logs).  Use this feature to remove `"Authorization"` header and `"access_token"` parameter from logs.

**Comment by nurali-techie** [comment]
Reference - https://docs.traefik.io/v1.6/configuration/logs/#access-logs

For `Authorization` below changes done in traefik toml configuration file.

[accessLog]
format = "json"
  [accessLog.fields.headers]
  defaultMode = "keep"
    [accessLog.fields.headers.names]
    "Authorization" = "redact"

There were two possibility, either `drop` or `redact`.
drop will drop the field from logs altogether.
redact will keep the field but the value of field will always be "REDACTED"

redact is better choice as we set "Authorization" in our code.  If there are any bug in code due to which "Authorization" is not get set then we can catch such bug when using  redact.

Before change:
```
{"BackendAddr":"api.starter-us-east-2a.openshift.com","BackendName":"backend1","BackendURL":{"Scheme":"https","Opaque":"","User":null,"Host":"api.starter-us-east-2a.openshift.com","Path":"/","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""},"ClientAddr":"10.1.3.1:41982","ClientHost":"54.242.198.185","ClientPort":"41982","ClientUsername":"-","DownstreamContentSize":0,"DownstreamStatus":0,"DownstreamStatusLine":"000 ","Duration":185577002,"FrontendName":"frontend1","OriginContentSize":0,"OriginDuration":185044221,"OriginStatus":0,"OriginStatusLine":"000 ","Overhead":532781,"RequestAddr":"f8osoproxy-test-dsaas-preview.b6ff.rh-idev.openshiftapps.com","RequestContentSize":0,"RequestCount":56241,"RequestHost":"f8osoproxy-test-dsaas-preview.b6ff.rh-idev.openshiftapps.com","RequestLine":"GET /api/v1/namespaces/osio-ci-e2e-002-preview-che/pods/workspaceres3yn2y1j2n9h2f.dockerimage/exec?command=sh\u0026command=-c\u0026command=cat+%3E%3E+%2Ftmp%2Fbootstrapper%2Fconfig.json+%3C%3C+%27EOF%27%0A%2C%0AEOF\u0026container=container\u0026stderr=true HTTP/1.1","RequestMethod":"GET","RequestPath":"/api/v1/namespaces/osio-ci-e2e-002-preview-che/pods/workspaceres3yn2y1j2n9h2f.dockerimage/exec?command=sh\u0026command=-c\u0026command=cat+%3E%3E+%2Ftmp%2Fbootstrapper%2Fconfig.json+%3C%3C+%27EOF%27%0A%2C%0AEOF\u0026container=container\u0026stderr=true","RequestPort":"-","RequestProtocol":"HTTP/1.1","RetryAttempts":0,"StartLocal":"2018-06-29T01:44:22.196726532Z","StartUTC":"2018-06-29T01:44:22.196726532Z","level":"info","msg":"","request_Accept-Encoding":"gzip","request_Authorization":"Bearer xxx","request_Connection":"Upgrade","request_Forwarded":"for=54.242.198.185;host=f8osoproxy-test-dsaas-preview.b6ff.rh-idev.openshiftapps.com;proto=https","request_Sec-Websocket-Key":"xWg0WY1wxTdrXfnFo+ymtw==","request_Sec-Websocket-Version":"13","request_Target":"https://api.starter-us-east-2a.openshift.com/","request_Upgrade":"websocket","request_User-Agent":"okhttp/3.9.1","request_X-Forwarded-For": 
"54.242.198.185","request_X-Forwarded-Host":"f8osoproxy-test-dsaas-preview.b6ff.rh-idev.openshiftapps.com","request_X-Forwarded-Port":"443","request_X-Forwarded-Proto":"https","time":"2018-06-29T01:44:22Z"}
```

Check `"request_Authorization":"Bearer xxx"` where xxx is the actual token.

After:
```
{"BackendAddr":"api.starter-us-east-2a.openshift.com","BackendName":"backend1","BackendURL":{"Scheme":"https","Opaque":"","User":null,"Host":"api.starter-us-east-2a.openshift.com","Path":"/","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""},"ClientAddr":"10.1.29.1:53540","ClientHost":"10.1.29.1","ClientPort":"53540","ClientUsername":"-","DownstreamContentSize":8299,"DownstreamStatus":200,"DownstreamStatusLine":"200 OK","Duration":17543025,"FrontendName":"frontend1","OriginContentSize":8299,"OriginDuration":17194198,"OriginStatus":200,"OriginStatusLine":"200 OK","Overhead":348827,"RequestAddr":"f8osoproxy","RequestContentSize":0,"RequestCount":3741,"RequestHost":"f8osoproxy","RequestLine":"GET /api/v1/namespaces/osio-ci-e2e-001-preview-run/replicationcontrollers HTTP/1.1","RequestMethod":"GET","RequestPath":"/api/v1/namespaces/osio-ci-e2e-001-preview-run/replicationcontrollers","RequestPort":"-","RequestProtocol":"HTTP/1.1","RetryAttempts":0,"StartLocal":"2018-06-29T11:22:41.7372209Z","StartUTC":"2018-06-29T11:22:41.7372209Z","downstream_Cache-Control":"no-store","downstream_Content-Type":"application/json","downstream_Date":"Fri, 29 Jun 2018 11:22:41 GMT","level":"info","msg":"","origin_Cache-Control":"no-store","origin_Content-Type":"application/json","origin_Date":"Fri, 29 Jun 2018 11:22:41 GMT","request_Accept":"application/json, */*","request_Accept-Encoding":"gzip","request_Authorization":"REDACTED","request_Target":"https://api.starter-us-east-2a.openshift.com/","request_User-Agent":"wit/v1.7.3  (linux/amd64) kubernetes/$Format","time":"2018-06-29T11:22:41Z"}
```

Check `"request_Authorization":"REDACTED"`

**Comment by nurali-techie** [comment]
Changes for `Authorization` with HK - https://gitlab.cee.redhat.com/dtsd/housekeeping/issues/2089

**Comment by nurali-techie** [comment]
For `access_token`

access_token is part of query parameter, it will be something like.
/some/path?param1=value1&access_token=secret

This path holds by to field in traefik.  `RequestPath` and `RequestLine`.

With traefik 1.6.x, one can drop fields altogether.  Like one can stop logging `RequestPath` field for always but it's not possible to drop this fields only when `access_token` is present as one of the param.

In short, removing `access_token` parameter from logs is not possible with traefik 1.6.x.

In case we want to remove `access_token`, we need to change traefik code.  Somewhere here [link](https://github.com/fabric8-services/fabric8-oso-proxy/blob/master/middlewares/accesslog/logger.go#L249).
